lib/search/hex_client.ex

```diff
@@ -1,47 +1,59 @@
 defmodule Search.HexClient do
-  @api_url "https://hex.pm/api"
   @repo_url "https://repo.hex.pm"
 
   alias Search.HexClient
 
   def get_releases(package_name) when is_binary(package_name) do
-    case get("#{@api_url}/packages/#{package_name}") do
-      {:ok, %{status: 200, body: %{"releases" => releases}}} ->
+    case get("packages/#{package_name}", plugins: [ReqHex]) do
+      {:ok, %{status: 200, body: releases}} ->
         res =
-          for %{"has_docs" => has_docs, "version" => version} <- releases do
+          for %{version: version} <- releases do
             %HexClient.Release{
               package_name: package_name,
-              version: Version.parse!(version),
-              has_docs: has_docs
+              version: Version.parse!(version)
             }
           end
 
         {:ok, res}
 
-      {:ok, %{status: status}} ->
-        {:error, "HTTP #{status}"}
-
       err ->
         err
     end
   end
 
   def get_docs_tarball(
-        %HexClient.Release{has_docs: has_docs, package_name: package_name, version: version} =
+        %HexClient.Release{package_name: package_name, version: version} =
           _release
       ) do
-    if has_docs do
-      case get("#{@repo_url}/docs/#{package_name}-#{version}.tar.gz") do
-        {:ok, %{status: 200, body: body}} -> {:ok, body}
-        {:ok, %{status: status}} -> {:error, "HTTP #{status}"}
-        err -> err
-      end
-    else
-      {:error, "Package release has no documentation."}
+    case get("docs/#{package_name}-#{version}.tar.gz") do
+      {:ok, %{status: 200, body: body}} ->
+        {:ok, body}
+
+      err ->
+        err
     end
   end
 
-  defp get(url) do
-    Req.get(url, Application.get_env(:search, :hex_client_req_options, []))
+  defp get(resource, opts \\ []) do
+    {_req, res} =
+      opts
+      |> Keyword.merge(Application.get_env(:search, :hex_client_req_options, []))
+      |> Keyword.merge(url: "#{@repo_url}/#{resource}")
+      |> Req.new()
+      |> Req.Request.prepend_response_steps(
+        handle_errors: fn {req, res} ->
+          # Looks like ReqHex fails in zlib on non-200 codes, so these are handled here
+          case res do
+            %{status: 200} -> {req, res}
+            %{status: status} -> {Req.Request.halt(req), RuntimeError.exception("HTTP #{status}")}
+          end
+        end
+      )
+      |> Req.Request.run_request()
+
+    case res do
+      %RuntimeError{message: message} -> {:error, message}
+      other -> {:ok, other}
+    end
```

Comment:
if there's a network error we'd return `{:ok, %Mint.TransportError{}}` (Req.TransportError on upcoming Req v0.5) which I think is gonna be unexpected. My suggestion is to instead of handling the error message here, make the request using Req.request, and handle the error at the call site, call Exception.message(e).

```suggestion
      |> Req.request()
```

WDYT?